sql/2019/07_Performance/07_18.sql

```diff
@@ -0,0 +1,47 @@
+#standardSQL

```

Comment:
Ditto 07_17

Reply:
- 1. Modified as suggested to use array methods to simplify the UDF, added try/catch. 
- 2. Found a scenario in the dataset where the "render-blocking-resources" had an error message "Required TagsBlockingFirstPaint gatherer encountered an error: NO_FCP", causing the UDF to return a null (as there is no "details" element inside "render-blocking-resources").  The APPROX_QUANTILES function would ignore NULL values by default. 
- 3. The earlier UDF would have returned 0 in such cases, which would have been an incorrect interpretation. 
- 4. Found this when i was matching the results of the previous UDF with the modified UDF.

![image](https://user-images.githubusercontent.com/24447527/63209644-2caa5800-c101-11e9-8fc8-e7a2ae0b23f6.png)

Reply:
The associated URL (there appear to be more)
SELECT report FROM `httparchive.almanac.lighthouse_mobile_1k` WHERE url = 'http://alcaravea.com/'